libraries/state_history/include/eosio/state_history/serialization.hpp

```diff
@@ -35,11 +49,27 @@ struct history_context_wrapper {
 };
 
 template <typename P, typename T>
-history_context_wrapper<std::decay_t<P>, std::decay_t<T>> make_history_context_wrapper(const chainbase::database& db,
-                                                                                       const P& context, const T& obj) {
+struct history_context_wrapper_stateless {
```

Comment:
I think I would have made `history_context_wrapper_stateless` a base of `history_context_wrapper`. 
the advantage is that the functions accepting a `history_context_wrapper_stateless&` would also take a `history_context_wrapper&`, although I'm not sure if it would have helped a lot.